editor/src/clj/editor/recent_files.clj

```diff
@@ -106,4 +106,53 @@
 
 (defn some-recent [prefs workspace evaluation-context]
   (->> (ordered-resource+view-types prefs workspace evaluation-context)
-       (take 10)))
\ No newline at end of file
+       (take 10)))
+
+(defn- tab->resource [^Tab tab]
+  (some-> tab
+    (ui/user-data :editor.app-view/view)
+    (g/node-value :view-data)
+    second
+    :resource))
+
+(defn- serialize-open-tabs [app-view]
+  (let [editor-tabs-split ^SplitPane (g/with-auto-evaluation-context ec
+                                       (g/node-value app-view :editor-tabs-split ec))]
+    (mapv (fn [^TabPane tab-pane]
+            (mapv (fn [tab]
+                    [(resource/proj-path (tab->resource tab))
```

Comment:
`resource/proj-path` will throw if it receives a `nil` argument, but `some->` in `tab->resource` implies it might return nil. Probably we should replace `some->` with `->` and assert that the returned value is not `nil`.